client/src/core/format.ts

```diff
@@ -94,22 +94,33 @@ const result_percent_formatter = {
   }),
 };
 
-const compact = (precision, val, lang, options) => {
+interface OptionsProps {
+  precision: number;
+  raw: boolean;
+  noMoney: boolean;
+}
+
+const compact = (
+  precision: number,
+  val: number,
+  lang: LangType,
+  options: OptionsProps
+) => {
   precision = precision || 0;
 
-  const abbrev = {
+  const abbrev: { [key: number]: { [key: string]: string } } = {
```

Comment:
`key` should be `LangType` instead of `string`. Even better: create a generic `LangDict<T>` type of the form `{ en: T, fr: T}`